### DISCUSSION TITLE
How to load a image from a path

### DISCUSSION THREAD

**Discussion by Arthex1** [rust-skia/rust-skia#819] [Q&A] [answered]
Hello, I have recently started using Skia and was curious about how I could do this. 

```rust
let image = image::io::Reader::open("/home/arthex/Pictures/test.png")
    .unwrap()
    .decode()
    .unwrap();
let image = image.into_bytes();
let data = skia_safe::Image::from_encoded(unsafe { skia_safe::Data::new_bytes(&image) }).unwrap();
```
I tried this, but it doesn't seem to work. Thats all, thanks.

**Comment by pragmatrix** [comment, marked as answer]
Looks to me like this decodes the image twice, first the image library and then passes the raw bytes to skia-safe (which attempts to decode again), try to read the encoded image with Rust's fs:: functions. See here for an example that uses include_bytes: https://github.com/rust-skia/rust-skia/blob/143f9371d3dee160b8eb6353a0672c258269dd49/skia-org/src/main.rs#L159-L163
